cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -421,6 +434,8 @@ public Void call() throws Exception {
         SystemMetadataWriter systemMetadataWriter =
           new StreamSystemMetadataWriter(metadataStore, streamId, config, createTime, description);
         systemMetadataWriter.write();
+        // Grant All access to the stream created to the User
+        authorizer.grant(streamId.toEntityId(), authenticationContext.getPrincipal(), ImmutableSet.of(Action.ALL));
```

Comment:
this should happen as one of the first steps after ensuring authorization on line 384. This is so that there is less of a possibility of orphane entities. Also, to counter orphane privileges, before granting, there should be a revoke all on the entity. If an exception happens while creation, there should be a revoke again. This is per https://wiki.cask.co/display/CE/Authorization+-+CDAP+3.5#Authorization-CDAP3.5-AuthorizationPolicies. Also see reference implementation in DatasetInstanceService.